src/ui/components/HostPermissions/index.js

```diff
@@ -0,0 +1,170 @@
+/* @flow */
+import * as React from 'react';
+import { compose } from 'redux';
+
+import log from 'core/logger';
+import translate from 'core/i18n/translate';
+import type { I18nType } from 'core/types/i18n';
+import Permission from 'ui/components/Permission';
+
+/* eslint-disable no-continue */
+type Props = {|
+  i18n: I18nType,
+  permissions: Array<string>,
+|};
+
+const domainMessageType: 'domainMessageType' = 'domainMessageType';
+const siteMessageType: 'siteMessageType' = 'siteMessageType';
+const allUrlsMessageType: 'allUrlsMessageType' = 'allUrlsMessageType';
+
+type HostPermissionMessageType = 'domainMessageType' | 'siteMessageType';
+type PermissionMessageType = HostPermissionMessageType | 'allUrlsMessageType';
+
+type GetPermissionStringParams = {|
+  messageType: PermissionMessageType,
+  param?: string | number,
+  inOther?: boolean,
+|};
+
+type GenerateHostPermissionsParams = {|
+  permissions: Array<string>,
+  // eslint-disable-next-line react/no-unused-prop-types
+  messageType: HostPermissionMessageType,
+|};
+
+export class HostPermissionsBase extends React.Component<Props> {
+  getPermissionString({ messageType, param, inOther = false }: GetPermissionStringParams): string {
```

Comment:
I think maybe `multiple` would work if you want to be generic, `multipleSites` if you don't mind saying "sites" for domains+sites (my preferences, technically they're all sites), and `multipleDomainsOrSites` if you want to be painfully clear 😉

Reply:
I was going to make it `inXOther` because the messages are `Access your data in %(param)s other domains` and `Access your data on %(param)s other sites`, but I didn't like the way `inXOther` looked, so it became just `inOther`, which I admit is pretty meaningless.

I'll just go with `multiple`.